src/Analyser/NodeScopeResolver.php

```diff
@@ -1804,6 +1805,12 @@ static function (Node $node, Scope $scope) use ($nodeCallback): void {
 				$exprResult = $this->processExprNode($stmt, $stmt->expr, $scope, $nodeCallback, ExpressionContext::createDeep());
 				$impurePoints = $exprResult->getImpurePoints();
 			}
+		} elseif ($stmt instanceof InlineHTML) {
+			$hasYield = false;
+			$throwPoints = [];
+			$impurePoints = [
+				new ImpurePoint($scope, $stmt, 'inlineHtml', 'inlineHtml', true),
```

Comment:
Please test the error message when this is done inside `@phpstan-pure` function. It should be nicer than `inlineHtml`.

I'd even say that the identifier should be renamed to something like `betweenPhpTags` or something.